src/coreclr/hosts/inc/coreclrhost.h

```diff
@@ -27,7 +27,8 @@ CORECLR_HOSTING_API(coreclr_initialize,
 
 CORECLR_HOSTING_API(coreclr_shutdown,
             void* hostHandle,
-            unsigned int domainId);
+            unsigned int domainId,
```

Comment:
I don't think we can change the signature of the existing hosting API. There are other Microsoft projects like dotnet cli and possibly also community projects using the current APIs and this would break them. It seems that we need to create a new version of coreclr_shutdown with the added latchedExitCode and keep the old one unchanged. Named e.g. coreclr_shutdown_2.